lib/listen/record.rb

```diff
@@ -64,7 +64,10 @@ def _sub_tree(rel_path)
         if path == rel_path
           result.merge!(meta)
         else
-          sub_path         = path.sub(%r{\A#{rel_path}/?}, '')
+          parts = path.split("/")
+          next unless parts.include?(rel_path)
```

Comment:
Hmm, can `rel_path` ever have separators in it? If not, then this would always have to be true, wouldn't it, since we established a few lines up that it _starts with_ `rel_path`?

Reply:
Without this line, a bunch of (my new) tests fail.

Reply:
> If not, then this would always have to be true, wouldn't it, since we established a few lines up that it starts with `rel_path`?

No, because for example, `"appspec.yml"` starts with `"app"`, but `["appspec.yml"]` doesn't *include* `"app"`.

Reply:
Oh jeez, I missed that too. Can we get rid of all the confusing special cases and simplify down to
```ruby
    def _sub_tree(rel_path)
      @tree.each_with_object({}) do |(path, meta), result|
        parts = path.split(::File::SEPARATOR)
        if parts.shift == rel_path
          if parts.empty?
            result.merge!(meta)
          else
            result[parts.join(::File::SEPARATOR)] = meta
          end
        end
      end
```

Reply:
Okay, I've moved to this implementation.